src/cli/repl.rs

```diff
@@ -0,0 +1,140 @@
+use std::{
+    env,
+    io::ErrorKind,
+    path::PathBuf,
+    process::{exit, ExitCode},
+};
+
+use anyhow::Error;
+use clap::Command;
+use lune::lua::stdio::formatting::pretty_format_luau_error;
+use lune::Lune;
+use mlua::ExternalError;
+use once_cell::sync::Lazy;
+use rustyline::{error::ReadlineError, history::FileHistory, DefaultEditor, Editor};
+
+fn env_var_bool(value: String) -> Option<bool> {
+    match value.to_lowercase().as_str() {
+        "true" | "1" => Some(true),
+        "false" | "0" => Some(false),
+        &_ => None,
+    }
+}
+
+// Isn't dependency injection plain awesome?!
+pub async fn show_interface(cmd: Command) -> Result<ExitCode, Error> {
+    let lune_version = cmd.get_version();
+
+    // The version is mandatory and will always exist
+    println!("Lune v{}", lune_version.unwrap());
+
+    let lune_instance = Lune::new();
+
+    let mut repl = DefaultEditor::new()?;
+
+    match repl.load_history(&(|| -> PathBuf {
+        let dir_opt = home::home_dir();
+
+        if let Some(dir) = dir_opt {
+            dir.join(".lune_history")
+        } else {
+            eprintln!("Failed to find user home directory, abort!");
+            // Doesn't feel right to exit directly with a exit code of 1
+            // Lmk if there is a better way of doing this
+            exit(1);
+        }
+    })()) {
+        Ok(_) => (),
+        Err(err) => {
+            match err {
+                err => {
+                    if let ReadlineError::Io(io_err) = err {
+                        if io_err.kind() == ErrorKind::NotFound {
+                            std::fs::write(
+                                // We know for sure that the home dir already exists
+                                home::home_dir().unwrap().join(".lune_history"),
+                                String::new(),
+                            )?;
+                        }
+                    }
+                }
+            }
+
+            eprintln!("WARN: Failed to load REPL history")
+        }
+    };
+
+    let mut interrupt_counter = 0u32;
+
+    let colorize: Lazy<bool> = Lazy::new(|| {
+        let no_color = env::var("NO_COLOR").unwrap_or_else(|_| "false".to_string());
+
+        if no_color.is_empty() {
+            true
+        } else {
+            !env_var_bool(no_color).unwrap_or_else(|| false)
+        }
+    });
```

Comment:
I think we can probably just default to always having colors on instead of this env var. The flag in `pretty_format_luau_error` is there to force colorization to be off in cases like stringifying errors from within lua (e.g. `tostring`), but for the CLI the `console` library we use for colorization will check for interactive terminals etc. and handle colors on/off there in a standardized way

Reply:
Ah, that's great.